Tests/LibC/TestLibCGrp.cpp

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2018-2023, the SerenityOS developers.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <LibTest/Macros.h>
+#include <LibTest/TestCase.h>
+
+#include <grp.h>
+
+void check_correctness(struct group* gr);
+
+void check_correctness(struct group* gr)
+{
+    auto member_count = [](char** gr_mem) {
+        size_t count = 0;
+        for (char** mem = gr_mem; *mem; ++mem) {
+            count++;
+        }
+        return count;
+    };
+    EXPECT_NE(gr, nullptr);
+    EXPECT_EQ(gr->gr_gid, (gid_t)3);
+    EXPECT_EQ(gr->gr_name, "phys"sv);
+    EXPECT_EQ(gr->gr_passwd, "x"sv);
+    EXPECT_EQ(member_count(gr->gr_mem), 2u);
+    EXPECT_EQ(gr->gr_mem[0], "window"sv);
+    EXPECT_EQ(gr->gr_mem[1], "anon"sv);
```

Comment:
This test feels a bit fragile, in that if we add more users to the phys group this test might break. 

It might be better in a future PR to invent a "list has member" type test helper to avoid the ordering requirement.

Reply:
thank you for reviewing, let me try checking it without the index